kubernetes/cmsweb/scripts/test_k8s_aps.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+##H Usage: test_k8s_aps.sh <base_url>
+##H
+
+if [ "$1" == "-h" ] || [ "$1" == "-help" ] || [ "$1" == "--help" ] || [ "$1" == "help" ] || [ "$1" == "" ]; then
+    perl -ne '/^##H/ && do { s/^##H ?//; print }' < $0
+    exit 1
+fi
+
+X509_USER_KEY=$HOME/.globus/userkey.pem
+X509_USER_CERT=$HOME/.globus/usercert.pem
+
+base_url=$1
+
+# Function to perform the request with automatic redirect following
+perform_request() {
+    local target_url="$1"
+    final_status_code=$(curl -L -i --key $X509_USER_KEY --cert $X509_USER_CERT -o /dev/null -w '%{http_code}' -s "$target_url")
+    echo "$target_url HTTP code: $final_status_code"
+}
+
+# CouchDB
```

Comment:
You have repeated pattern, i.e. you execute `perform_request` over provided list of URL end-points. So, generalize the script and remove individual lists. This way the script will be much smaller and it can take external file (list of end-points). We'll keep such list separate from the file and can adjust it as necessary, but the script itself will stay intact and only take input list.